docs/2-configuration.md

```diff
@@ -1,7 +1,124 @@
 # Configuration
 
-Emerauth (specifically this repository) is configured from the ground up based on a `config.edn` resource. You can see a simple example [here](./resources/config.example.dn). The entire configuration can be overwhelming, but bear with us, we'll try to make it all understandable.
+Emerauth is configured from the ground up based on a `config.edn` resource. You can see a simple example [here](./resources/config.example.dn). The entire configuration can be overwhelming, but bear with us, we'll try to make it all understandable.
 
-## Resources
+## Resource
 
-## Apps
+A resource is anything that can be authorized and then accessed. A resource is identified by a name, and then matched with its options. For example:
+
+```clj
+{"/auth/*" {:methods ["get"]}}
+```
+
+This declares a resource named `/auth/*` that can only be authorized when the HTTP method is `GET`.
+
+> [!WARNING]
+> The resource `/auth/*` may look like a URI resource, but it has nothing to do with HTTP. It is just an identifier, a literal string that can be later used to set it apart from other resources.
+
+A resource is not bound to a protocol. As long as its identifier (name) and its options match, they can define HTTP resources, gRPC resources, graphQL resources, etc.
+
+## Resource Group
+
+These resources by themselves cannot do much. We need to group them so we can identify what user context they're part of. For example, the following resources:
+
+```clj
+{"/auth" {:methods ["post" "put"]}
+ "/home" {:methods ["get"]}
+ "/store" {:methods ["get"]}
+ "/store/:product" {:methods ["get"]}
+ "/marketplace" {:methods ["get"]}
+ "/marketplace/:product" {:methods ["get" "post" "put" "delete"]}}
+```
+
+can be grouped and contextualized as:
+
+```clj
+{:public {"/auth" {:methods ["post"]}
+          "/home" {:methods ["get"]}}
+
+ :protected {"/auth" {:methods ["put"]}
+             "/store" {:methods ["get"]}}
+
+ :marketplace {"/marketplace" {:methods ["get"]}
+               "/marketplace/:product" {:methods ["get" "post" "put" "delete"]}}}
+```
+
+The `:public` and `:protected` resource groups are registered keywords for Emerauth. They are, respectively, resources that anyone can access, and resources that only signedin users can access.
```

Comment:
Fix typo: "signed in" instead

Reply:
Since these changes aren't trivial, please make a change suggestion! Valid for the other reviewed section as well.

![image](https://github.com/emerauth/emerauth/assets/41875891/b3817723-7c21-4389-b5af-eb6f9f52a6b3)

Anyways, thanks for the review!

Reply:
Also, when doing reviews on multiple files, please don't send each one separately. Write your review for each problem you encounter, and then send everything as a single review.